### PR TITLE
feat: improve error logs format

### PR DESCRIPTION
**Pull request by mrednic-1A**
## Proposed change

Improve error log messages

### CODE REVIEW

**Review by matthieu-crouzet — APPROVED** [review]

File: packages/@ama-sdk/core/src/fwk/errors.ts [hunk 1]
```diff
@@ -16,10 +18,10 @@ interface GenericErrorContext {
 export class GenericError extends Error {
 
   constructor(message: string, context?: GenericErrorContext) {
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/errors.ts:20]: Maybe you should turn the `context` field as `public` to have it in the error serialization

File: packages/@ama-sdk/core/src/fwk/api.helpers.ts [hunk 2]
```diff
@@ -115,3 +115,24 @@ export function tokenizeRequestOptions(tokenizedUrl: string, queryParameters: {
...
 }
+
+/**
+ * Remove http/https from the given url
+ * Extract origin part from the given url
+ * If no url return 'unknown' string;
+ *
+ * @param url The string url
+ */
+export function extractInfoFromUrl(url?: string | null) {
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/api.helpers.ts:126]: Would not work to use the ES6 URL object?

```typescript
if (url) {
  const parsedUrl = new URL(url);
  return parsedURL.hostname + parsedURL.pathname;
}

return 'unknown';

File: packages/@ama-sdk/core/src/fwk/api.helpers.spec.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import { extractInfoFromUrl } from './api.helpers';
+
+describe('ExtractInfoFromUrl', () => {
```
Comment (kpanot) [inline packages/@ama-sdk/core/src/fwk/api.helpers.spec.ts:3]: your forgot to test your supported Basic Auth URL

File: packages/@ama-sdk/core/src/fwk/errors.ts (hunk 1 above)
Comment (mrednic-1A) [reply packages/@ama-sdk/core/src/fwk/errors.ts:20]: I have done some changes from the previous implementation, and it does not seem to be needed to be serialized.

File: packages/@ama-sdk/core/src/fwk/api.helpers.spec.ts (hunk 3 above)
Comment (mrednic-1A) [reply packages/@ama-sdk/core/src/fwk/api.helpers.spec.ts]: I removed the factory function and the test too. I found another way to get the origin from the `headers` directly

**Review by matthieu-crouzet — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Review by pginoux-1A — APPROVED** [review]

**Pull request merged**